pokemon-rest/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/controllers/PokemonSpeciesController.java

```diff
@@ -0,0 +1,147 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.controllers;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.ApiUris;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.ChangePreevolutionDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.ChangeTypingDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.PokemonSpeciesCreateDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.dto.PokemonSpeciesDTO;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.CircularEvolutionChainException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.EntityIsUsedException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.EntityUsedException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.EvolutionChainTooLongException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.InvalidParameterException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.InvalidPreevolutionChangeException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.exceptions.ResourceNotFoundException;
+import cz.muni.fi.pa165.pokemon.league.participation.manager.facade.PokemonSpeciesFacade;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.List;
+import javax.inject.Inject;
+import javax.validation.Valid;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.RequestBody;
+
+/**
+ * This is the controller class of Pokemon Species to handle all the calls of rest api.
+ * 
+ * @author Tamás Rózsa 445653
+ */
+@RestController
+@RequestMapping(ApiUris.POKEMON_SPECIES_URI)
+public class PokemonSpeciesController {
+
+    @Inject
+    private PokemonSpeciesFacade pokemonSpeciesFacade;
+    
+    @RequestMapping(
+            value = "/changeTyping",
+            method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public void changeTyping(@Valid @RequestBody ChangeTypingDTO newTyping) {
+        try {
+            pokemonSpeciesFacade.changeTyping(newTyping);
+        }
+        catch(Exception ex) {
+            throw new ResourceNotFoundException();
```

Comment:
Either include ```ex``` as reason for ```ResourceNotFoundException```, or at the very least log it with a reasonable log level (e.g. DEBUG).